sql/mysql/migrate.go

```diff
@@ -52,6 +54,69 @@ func (p *planApply) ApplyChanges(ctx context.Context, changes []schema.Change) e
 	return sqlx.ApplyChanges(ctx, changes, p)
 }
 
+// A revReadWrite provides implementation for the migrate.RevisionReadWriter interface.
+type revReadWrite struct {
+	c *ent.Client
+	// sc is the function signature of the Ent migration engine.
+	// Due to cyclic dependencies between Ent and Atlas, this value is stitched in at runtime.
+	sc func(context.Context) error
+}
+
+// ReadRevisions read the revisions from the revisions table.
+func (r *revReadWrite) ReadRevisions(ctx context.Context) (migrate.Revisions, error) {
+	rs, err := r.c.QueryContext(ctx, fmt.Sprintf("SHOW TABLES LIKE '%s'", revision.Table))
+	if err != nil {
+		return nil, err
+	}
+	if !rs.Next() {
+		// If there are no results the table does not exist and there are no revisions yet.
+		return migrate.Revisions{}, nil
+	}
+	if err := rs.Close(); err != nil {
+		return nil, err
+	}
+	revs, err := r.c.Revision.Query().All(ctx)
```

Comment:
I think we need to add some ORDERing here. There's no default ordering in the database.

Reply:
Well, there is: order of insertion. But to be safe we can add one.

Reply:
There’s not guarantee that rows will return some order unless the ORDER BY clause is used.

Reply:
There’s no guarantee that rows will return in some order unless the ORDER BY clause is used.